src/Console/Command/WorkerCommand.php

```diff
@@ -0,0 +1,243 @@
+                        $out->write([
+                            'action' => ParallelAction::WORKER_ERROR_REPORT,
+                            'message' => $error->getMessage(),
+                            'file' => $error->getFile(),
+                            'line' => $error->getLine(),
+                            'code' => $error->getCode(),
+                            'trace' => $error->getTraceAsString(),
+                        ]);
+                    };
+                    $out->on('error', $handleError);
+                    $in->on('error', $handleError);
+
+                    // [REACT] Listen for messages from the parallelisation operator (analysis requests)
+                    $in->on('data', function (array $json) use ($loop, $runner, $out): void {
+                        $action = $json['action'] ?? null;
+
+                        // Parallelisation operator does not have more to do, let's close the connection
+                        if (ParallelAction::RUNNER_THANK_YOU === $action) {
+                            $loop->stop();
+
+                            return;
+                        }
+
+                        if (ParallelAction::RUNNER_REQUEST_ANALYSIS !== $action) {
+                            // At this point we only expect analysis requests, if any other action happen, we need to fix the code.
+                            throw new \LogicException(sprintf('Unexpected action ParallelAction::%s.', $action));
+                        }
+
+                        /** @var iterable<int, string> $files */
+                        $files = $json['files'];
+
+                        foreach ($files as $absolutePath) {
+                            // Reset events because we want to collect only those coming from analysed files chunk
+                            $this->events = [];
+                            $runner->setFileIterator(new \ArrayIterator([new \SplFileInfo($absolutePath)]));
+                            $analysisResult = $runner->fix();
+
+                            if (1 !== \count($this->events)) {
+                                throw new ParallelisationException('Runner did not report a fixing event or reported too many.');
+                            }
+
+                            if (1 < \count($analysisResult)) {
+                                throw new ParallelisationException('Runner returned more analysis results than expected.');
+                            }
+
+                            $out->write([
+                                'action' => ParallelAction::WORKER_RESULT,
+                                'file' => $absolutePath,
+                                'fileHash' => $this->events[0]->getFileHash(),
+                                'status' => $this->events[0]->getStatus(),
+                                'fixInfo' => array_pop($analysisResult),
+                                'errors' => $this->errorsManager->forPath($absolutePath),
```

Comment:
For events: we do `->events = []` above, which is clearing the events array, knowing that for fixing single file we have the events array with data related only to this single file.

For errors: we do `->forPath(path)`, so with every iteration over next and next file, we need to filter errors related to single path from larger and larger array.
With `->popAllErrors()` approach, you mimic the `->events=[]` approach and have no need to filter the errors at all, as you know it will always contain array for single file from the loop, only.

Reply:
This is edge case because errors rather don't occur often, we can improve it later if needed.

Reply:
even more, why not to take commit i provided ?

Reply:
I provided the reasoning [here](https://github.com/Wirone/PHP-CS-Fixer/pull/5#discussion_r1592772034) - the approach that pops all errors is "not compatible" with running `$out->write()` outside of the file chunk loop. Yes, we could probably pop all errors inside a loop and store them in temporary variable to report them outside of the loop, but requesting for errors collected for specific file is more explicit and I like it better.

Reply:
yes, in loop, that was my suggestion. we already get those errors inside the loop, not outside of loop. what you mean to be not compatible here?

you report each file in loop, you extract the events (and cut off the already consumed ones), we can do same for errors. no